### PR TITLE
Fix a11y on amp-pinterest

### PR DESCRIPTION
**Pull request by cathyxz**
Closes https://github.com/ampproject/amphtml/issues/19281. 

The more severe problem here is the untabbable and unclickable `Save` button that's a `<span>` in `pin-widget.js`. Addressed that by adding a `tabindex`, `aria-label`, and a `onKeyDown` handler that triggers the click handler if `SPACE` or `ENTER` while selected. 

Ran lighthouse a few times and added aria labels and such--we should probably do this more systematically with our components.

### CODE REVIEW

File: extensions/amp-pinterest/0.1/pin-widget.js [hunk 1]
```diff
@@ -293,6 +306,8 @@ export class PinWidget {
 
     // listen for clicks
     structure.addEventListener('click', this.handleClick.bind(this));
+    // Handle Space and Enter while selected for a11y
+    structure.addEventListener('keydown', this.handleKeyDown.bind(this));
```
Comment (sparhami) [inline extensions/amp-pinterest/0.1/pin-widget.js]: Change this to `keypress`. A real button only performs an action when you actually release the key.

If it isn't too much trouble, also consider just using a real button.

File: extensions/amp-pinterest/0.1/pin-widget.js [hunk 2]
```diff
@@ -188,6 +200,7 @@ export class PinWidget {
           'img': {
             'className': '-amp-pinterest-embed-pin-text-icon-attrib',
             'src': pin['attribution']['provider_icon_url'],
+            'alt': 'Pinterest image attribution icon',
```
Comment (sparhami) [inline extensions/amp-pinterest/0.1/pin-widget.js]: Hmm, does this really provide the right (or useful) context to a screen reader user? For example, if this was `aria-hidden` or had an empty `alt`, the content of the attribution would be `by <author name>`, which would I think would get the point across.

One thing to keep an eye out for on a11y is to make sure we don't spam the user with too much info. For a sighted user, it is easy enough to gloss over the icon, but a screen reader user will have to move past it and at least partially listen to it to see if it is useful.

It seems by omitting this, we would probably make the overall experience better.

Comment (cathyxz) [reply extensions/amp-pinterest/0.1/pin-widget.js]: Based on their JSON format: 
```
attribution: {title: "Next stop: Pinterest!", url: "http://www.flickr.com/photos/kentbrew/6851755809/",…}
author_name: "kentbrew"
author_url: "http://www.flickr.com/photos/kentbrew/"
provider_favicon_url: "https://s.pinimg.com/images/api/attrib/flickr.png"
provider_icon_url: "https://s.pinimg.com/images/api/attrib/flickr@2x.png"
provider_name: "flickr"
title: "Next stop: Pinterest!"
url: "http://www.flickr.com/photos/kentbrew/6851755809/"
```

It does look like they have a `provider_name` attribute that we can use to provide some useful information. Changed to include that.

File: extensions/amp-pinterest/0.1/pin-widget.js (hunk 1 above)
Comment (cathyxz) [reply extensions/amp-pinterest/0.1/pin-widget.js]: Changed to keypress. I think I need to take a closer look at the UI per https://github.com/ampproject/amphtml/issues/23116 anyway, so I'll look at the button thing along with the rest of the UI in a following PR.

**Pull request merged**